source/NVDAObjects/IAccessible/mozilla.py

```diff
@@ -73,6 +74,17 @@ def _get_detailsSummary(self) -> Optional[str]:
 			# just take the first for now.
 			return target.summarizeInProcess()
 
+	def _get_detailsRole(self) -> Optional[controlTypes.Role]:
+		# Unlike base Ia2Web implementation, the details-roles
+		# IA2 attribute is not exposed in FireFox.
+		# Although slower, we have to fetch the details relations instead.
+		detailsRelations = self.detailsRelations
+		if not detailsRelations:
+			return None

```

Comment:
This relies on `detailsRelations` observing `config.conf["debugLog"]["annotations"]`. Easy for that to change and break this.

Reply:
this is for debug logging.

`config.conf["annotations"]["reportDetails"]` is checked in `speech.py` and `braille.py`

Reply:
ah yes, I must have misread that.